tensorflow/compiler/jit/xla_gpu_device.cc

```diff
@@ -53,20 +55,21 @@ Status XlaGpuDeviceFactory::CreateDevices(
     VLOG(1) << "Failed to create XLA_GPU device: " << platform.status();
     return Status::OK();
   }
-  const auto& allowed_gpus =
+  string allowed_gpus =
       session_options.config.gpu_options().visible_device_list();
-  std::unordered_set<int> gpu_ids;
+  std::set<int> gpu_ids;
   int num_visible_devices = platform.ValueOrDie()->VisibleDeviceCount();
   if (allowed_gpus.empty()) {
     for (int i = 0; i < num_visible_devices; ++i) gpu_ids.insert(i);
   } else {
+    // For loop below is copied from gpu/gpu_device.cc. It validates
+    // configuration string. It should be redundant since code would fail there
+    // before it gets to here.
```

Comment:
Sorry to belabor this, but the for loop isn't redundant, because it's also what inserts elements into gpu_ids.